Shared/ModuleHandle.ps1

```diff
@@ -0,0 +1,85 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+<#
+.SYNOPSIS
+This script defines a function `Request-Module` that checks for the presence of specified PowerShell modules.
+If a module is not found, it attempts to install it on current user scope.
+The function accepts a list of module names and an optional minimum version for the modules.
+It returns a boolean indicating whether all specified modules are installed successfully.
+
+.PARAMETER Modules
+ Mandatory array of strings specifying the names of the modules to check and install if necessary.
+.PARAMETER MinModuleVersion
+ Optional parameter to specify the minimum version of the modules (default is null).
+
+.OUTPUTS
+bool. A boolean indicating whether all specified modules are installed successfully.
+
+.EXAMPLE
+$requestModule = Request-Module -Modules "ExchangeOnlineManagement"
+This example checks if the "ExchangeOnlineManagement" module is installed. If it is not found, the script attempts to install it.
+
+.EXAMPLE
+$requestModule = Request-Module -Modules "ExchangeOnlineManagement" -MinModuleVersion $MinModuleVersion
+This example checks if the "ExchangeOnlineManagement" module with a specified minimum version is installed. If it is not found, the script attempts to install it.
+#>
+
+function Request-Module {
+    [OutputType([bool])]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string[]]$Module,
+        [Parameter(Mandatory = $false)]
+        [System.Version]$MinModuleVersion = $null
+    )
+
+    $noFoundError = $true
+    foreach ($m in $Module) {
+        Write-Verbose "Checking $m PowerShell Module"
+        $getParams = @{
+            Name        = $m
+            ErrorAction = 'SilentlyContinue'
+        }
+        if ($MinModuleVersion) {
+            $getParams["MinimumVersion"] = $MinModuleVersion
+            Write-Verbose "with minimum version $minModuleVersion"
+        } else {
+            Write-Verbose "without minimum version"
+        }
+        $installed = Get-InstalledModule @getParams
+
+        if ($null -eq $installed -or $installed.Name -notcontains $m) {
+            Write-Host "The following module is missing: $m" -ForegroundColor Yellow
+            $confirmed = $null
+            try {
+                Write-Verbose "Installing $m"
+                $installParams = @{
+                    Name        = $m
+                    Scope       = "CurrentUser"
```

Comment:
Not sure if this could be useful in future, can we make the scope adjustable via parameter (`CurrentUser` could be set as default here). What do you think?

Reply:
I use CurrentUser because if we use Alluser it requires elevation and we will need to check it before.
in the current scripts we do not need elevation to have the script working and that was the main reason to avoid Alluser, but if you think it could be useful for other scripts I can include as a parameter.

Reply:
I think there are use cases in the future but I'm okay with keeping the current approach for the first version of this script.